src/libs/WorkspacesSettingsUtils.ts

```diff
@@ -211,6 +212,129 @@ function getUnitTranslationKey(unit: Unit): TranslationPaths {
+                return null;
+        }
+    };
+
+    const confirmationButtonText = () => {
+        switch (error) {
+            case CONST.POLICY.OWNERSHIP_ERRORS.AMOUNT_OWED:
+                return translate('workspace.changeOwner.amountOwedButtonText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.OWNER_OWES_AMOUNT:
+                return translate('workspace.changeOwner.ownerOwesAmountButtonText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.SUBSCRIPTION:
+                return translate('workspace.changeOwner.subscriptionButtonText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.DUPLICATE_SUBSCRIPTION:
+                return translate('workspace.changeOwner.duplicateSubscriptionButtonText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.HAS_FAILED_SETTLEMENTS:
+                return translate('workspace.changeOwner.hasFailedSettlementsButtonText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.FAILED_TO_CLEAR_BALANCE:
+                return translate('workspace.changeOwner.failedToClearBalanceButtonText');
+            default:
+                return '';
+        }
+    };
+
+    const confirmationText = () => {
+        const changeOwner = policy?.errorFields?.changeOwner;
+        const subscription = changeOwner?.subscription as unknown as {ownerUserCount: number; totalUserCount: number};
+
+        switch (error) {
+            case 'noBillingCard':
+                break;
+            case CONST.POLICY.OWNERSHIP_ERRORS.AMOUNT_OWED:
+                return translate('workspace.changeOwner.amountOwedText');
+            case CONST.POLICY.OWNERSHIP_ERRORS.OWNER_OWES_AMOUNT:
+                return translate('workspace.changeOwner.ownerOwesAmountText', {
+                    email: changeOwner?.ownerOwesAmount,
+                    amount: '',
+                });
+            case CONST.POLICY.OWNERSHIP_ERRORS.SUBSCRIPTION:
+                return translate('workspace.changeOwner.subscriptionText', {
+                    usersCount: subscription?.ownerUserCount,
+                    finalCount: subscription?.totalUserCount,
+                });
+            case CONST.POLICY.OWNERSHIP_ERRORS.DUPLICATE_SUBSCRIPTION:
+                return translate('workspace.changeOwner.duplicateSubscriptionText', {
+                    email: changeOwner?.duplicateSubscription,
+                    workspaceName: policy?.name,
+                });
+            case CONST.POLICY.OWNERSHIP_ERRORS.HAS_FAILED_SETTLEMENTS:
+                return translate('workspace.changeOwner.hasFailedSettlementsText', {email: accountLogin});
+            case CONST.POLICY.OWNERSHIP_ERRORS.FAILED_TO_CLEAR_BALANCE:
+                return translate('workspace.changeOwner.failedToClearBalanceText');
+            default:
+                return null;
+        }
+    };
+
+    const title = confirmationTitle();
+    const text = confirmationText();
+    const buttonText = confirmationButtonText();
+
+    return {title, text, buttonText};
+}
```

Comment:
Hmm I think what I had in mind for this was something like:
```suggestion
function getOwnershipChecksDisplayText(
    error: ValueOf<typeof CONST.POLICY.OWNERSHIP_ERRORS>,
    translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: PhraseParameters<Phrase<TKey>>) => string,
    policy: OnyxEntry<Policy>,
    accountLogin: string | undefined,
) {
    let title;
    let text;
    let buttonText;

    const changeOwner = policy?.errorFields?.changeOwner;
    const subscription = changeOwner?.subscription as unknown as {ownerUserCount: number; totalUserCount: number};

    switch (error) {
        case CONST.POLICY.OWNERSHIP_ERRORS.AMOUNT_OWED:
            title = translate('workspace.changeOwner.amountOwedTitle');
            text = translate('workspace.changeOwner.amountOwedText');
            buttonText = translate('workspace.changeOwner.amountOwedButtonText');
            break;
        case CONST.POLICY.OWNERSHIP_ERRORS.OWNER_OWES_AMOUNT:
            title = translate('workspace.changeOwner.ownerOwesAmountTitle');
            text = translate('workspace.changeOwner.ownerOwesAmountText', {email: changeOwner?.ownerOwesAmount, amount: ''});
            buttonText = translate('workspace.changeOwner.ownerOwesAmountButtonText');
            break;
        case CONST.POLICY.OWNERSHIP_ERRORS.SUBSCRIPTION:
            title = translate('workspace.changeOwner.subscriptionTitle');
            text = translate('workspace.changeOwner.subscriptionText', {usersCount: subscription?.ownerUserCount, finalCount: subscription?.totalUserCount});
            buttonText = translate('workspace.changeOwner.subscriptionButtonText');
            break;
        case CONST.POLICY.OWNERSHIP_ERRORS.DUPLICATE_SUBSCRIPTION:
            title = translate('workspace.changeOwner.duplicateSubscriptionTitle');
            text = translate('workspace.changeOwner.duplicateSubscriptionText', {email: changeOwner?.duplicateSubscription, workspaceName: policy?.name});
            buttonText = translate('workspace.changeOwner.duplicateSubscriptionButtonText');
            break;
        case CONST.POLICY.OWNERSHIP_ERRORS.HAS_FAILED_SETTLEMENTS:
            title = translate('workspace.changeOwner.hasFailedSettlementsTitle');
            text = translate('workspace.changeOwner.hasFailedSettlementsText', {email: accountLogin});
            buttonText = translate('workspace.changeOwner.hasFailedSettlementsButtonText');
            break;
        case CONST.POLICY.OWNERSHIP_ERRORS.FAILED_TO_CLEAR_BALANCE:
            title = translate('workspace.changeOwner.failedToClearBalanceTitle');
            text = translate('workspace.changeOwner.failedToClearBalanceText');
            buttonText = translate('workspace.changeOwner.failedToClearBalanceButtonText');
            break;
        default:
            title = '';
            buttonText = '';
            text = '';
            break;
    }

    return {title, text, buttonText};
}
```

Reply:
changed